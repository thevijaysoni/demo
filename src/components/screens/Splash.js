import React, { useEffect } from 'react';
import { Image, StyleSheet, View } from 'react-native';
import { spacing } from '../../styles/spacing';
import colors from '../../utility/Colors';
import { navigate } from '../../utility/commonFunctions';
import { KEY_APP } from '../../utility/constants';
import { Images } from '../../utility/imageRes';

const Splash = ({ }) => {

    useEffect(() => {
        navigate(KEY_APP)
        // setTimeout(() => {
        // }, 2000);
    }, [])

    return (
        <View style={{ flex: 1, backgroundColor: colors.white, justifyContent: 'center', alignItems: 'center' }} >
            <Image
                source={Images.IMG_APP_LOGO}
                style={styles.logoStyle}
                resizeMode={'contain'}
            />
        </View>
    );
}
const styles = StyleSheet.create({
    logoStyle: {
        height: spacing.FULL_WIDTH / 3,
        width: spacing.FULL_WIDTH / 3,
    }
});

export default Splash;