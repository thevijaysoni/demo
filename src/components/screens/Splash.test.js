import React from 'react';
import renderer from 'react-test-renderer';
import Splash from './Splash';
import { shallow, configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import { useEffect } from 'react';

configure({ adapter: new Adapter() });
describe("Splash", () => {

    beforeEach(() => {

    });

    it('should render splash screen correctly', () => {
        const tree = renderer.create(<Splash />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should call useEffect', () => {
        const prop = {
            navigate: jest.fn()
        }
        let component = shallow(<Splash {...prop} />);

        expect(prop.navigate).toHaveBeenCalled();
    })

    it('should navigate to app navigator after 2 seconds of useEffect calling', () => {

    })
});