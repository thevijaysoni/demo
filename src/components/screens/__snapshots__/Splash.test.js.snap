// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash should render splash screen correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../src/assets/images/logo.png",
      }
    }
    style={
      Object {
        "height": 250,
        "width": 250,
      }
    }
  />
</View>
`;
